### PR TITLE
PowerShell Remoting exploiter refactor

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do? 

Addresses part of #1246. Refactors the existing powershell exploiter.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing? 
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running one case of the exploiter on a test machine. See screenshots.
* [x] If applicable, add screenshots or log transcripts of the feature working

image
---
image

### CODE REVIEW

File: monkey/common/utils/exceptions.py [hunk 1]
```diff
@@ -6,7 +6,11 @@ class FailedExploitationError(Exception):
     """ Raise when exploiter fails instead of returning False """
 
 
-class InvalidRegistrationCredentialsError(Exception):
+class CredentialsError(Exception):
+    """ Raise when credentials are wrong"""
+
+
```
Comment (mssalvatore) [inline monkey/common/utils/exceptions.py:12]: This is never used, except for `InvalidRegistrationCredentialsError` inherits it.

File: monkey/infection_monkey/dropper.py [hunk 2]
```diff
@@ -202,7 +202,7 @@ def cleanup(self):
                     )
 
                     # mark the file for removal on next boot
-                    dropper_source_path_ctypes = c_char_p(self._config["source_path"])
+                    dropper_source_path_ctypes = c_char_p(self._config["source_path"].encode())
```
Comment (mssalvatore) [inline monkey/infection_monkey/dropper.py:205]: I think the hadoop exploiter manifests this bug. We should test it.

File: monkey/infection_monkey/exploit/powershell.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    def __init__(self, host: VictimHost):
+        PowerShellExploiter._set_sensitive_packages_log_level_to_error()
+
+        super().__init__(host)
+        self.client = None
+
+    @staticmethod
+    def _set_sensitive_packages_log_level_to_error():
+        # If root logger is inherited, extensive and potentially sensitive info could be logged
+        sensitive_packages = [pypsrp, spnego, connectionpool]
```
Comment (ilija-lazoroski) [inline monkey/infection_monkey/exploit/powershell.py:39]: Do we need to add these in the Pipfile ?

File: monkey/infection_monkey/exploit/powershell_utils/utils.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    credentials = []
+    credentials.extend(_get_empty_credentials(is_windows))
+    credentials.extend(_get_username_only_credentials(usernames, is_windows))
+    credentials.extend(_get_username_password_credentials(usernames, passwords))
+
+    return credentials
+
+
+def _get_empty_credentials(is_windows: bool) -> List[Tuple[None, None]]:
```
Comment (shreyamalviya) [inline monkey/infection_monkey/exploit/powershell_utils/utils.py:28]:
```suggestion
def _get_empty_credentials(is_windows: bool) -> Union[List[Tuple[None, None]], List[]]:
```

File: monkey/tests/unit_tests/infection_monkey/exploit/powershell_utils/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+def test_get_credentials__username_only_windows_true():
+    credentials = utils.get_credentials(TEST_USERS, [], True)
+
+    assert len(credentials) == 5
+    assert (TEST_USERS[0], "") in credentials
+    assert (TEST_USERS[1], "") in credentials
+    assert (TEST_USERS[0], None) in credentials
+    assert (TEST_USERS[1], None) in credentials
+
```
Comment (shreyamalviya) [inline monkey/tests/unit_tests/infection_monkey/exploit/powershell_utils/test_utils.py:37]: Let's switch the order of these two functions so we maintain the "windows_true" followed by "windows_false" pattern.

File: monkey/tests/unit_tests/infection_monkey/exploit/powershell_utils/test_utils.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+def test_get_credentials__username_password():
```
Comment (shreyamalviya) [inline monkey/tests/unit_tests/infection_monkey/exploit/powershell_utils/test_utils.py:39]: We should add a "_windows_true" at the end of this function's name just to be explicit.

**Pull request merged**